spec/lib/krikri/mapper_agent_spec.rb

```diff
@@ -0,0 +1,69 @@
+describe Krikri::Mapper::Agent do
+  opts = { name: :agent_map, generator_uri: 'http://example.org/agent/gen/1' }
+  it_behaves_like 'a software agent', opts, Krikri::MappingJob
+
+  subject { described_class.new(opts) }
+
+  let(:generator_uri) { 'http://example.org/agent/gen/1' }
+  let(:activity_uri) { 'http://example.org/agent/act/1' }
+  let(:mapping_name) { :agent_map }
+  let(:opts) { { name: mapping_name, generator_uri: generator_uri } }
+
+  describe '#run' do
+    let(:record_double) { double(DPLA::MAP::Aggregation) }
+    let(:records) do
+      [record_double, record_double.clone, record_double.clone]
+    end
+
+    before do
+      allow(subject).to receive(:records).and_return([:record1, :record2])
+      allow(record_double).to receive(:node?).and_return(true)
+      allow(record_double).to receive(:mint_id!)
+      allow(record_double).to receive(:save)
+      expect(Krikri::Mapper).to receive(:map).with(mapping_name, subject.records)
+        .and_return(records)
+    end
+
+    it 'calls mapper' do
+      subject.run
+    end
+
+    it 'sets generator' do
+      records.each do |rec|
+        statement = double
+        allow(RDF).to receive(:Statement)
+          .with(rec, RDF::PROV.wasGeneratedBy, activity_uri)
+          .and_return(statement)
+        expect(rec).to receive(:<<).with(statement)
+      end
+      subject.run(activity_uri)
+    end
+  end
+
+  # TODO: these tests are tied closely to implementation.
+  #       This is a code smell. Consider refactor of ProvenanceQueryClient
+  #       and #records.
+  describe '#records' do
+    let(:query) { double(RDF::Query) }
+    let(:solutions) { double(RDF::Query::Solutions) }
+    let(:solution) { double(RDF::Query::Solution) }
+
+    let(:record_double) { double(Krikri::OriginalRecord) }
+
+    it 'returns a lazy enum' do
+      expect(subject.records).to be_a Enumerator::Lazy
+    end
+
+    it 'gets records for generated by harvest_activity' do
+      expect_any_instance_of(Krikri::ProvenanceQueryClient)
+        .to receive(:find_by_activity).with(generator_uri).and_return(query)
+      expect(query).to receive(:execute).and_return(solutions)
```

Comment:
These lines mock out the modules you reference exactly to avoid being an integration test.  The problem here is that there's a long list of dependencies that any user of `ProvenanceQueryClient` takes on board.  

I think the solution must be to have that module insulite its clients from `RDF::Query`, `RDF::Query::Solutions` and `RDF::Query::Solution`?